pkg/resource/service.go

```diff
@@ -1,5 +1,7 @@
 package resource
 
+//go:generate mockery --name=ServiceInterface -r --case underscore --with-expecter --structname ResourceService  --filename=resource_service.go --output=../../mocks
```

Comment:
I think interface name could be called better (without `-Interface` suffix)

The naming could follow what is suggested in [`effective go`](https://go.dev/doc/effective_go#interface-names)

Reply:
Will take this up in separate refactoring PR